lib/chef/resource/chocolatey_installer.rb

```diff
@@ -0,0 +1,135 @@
+
+      property :ignore_proxy, [TrueClass, FalseClass], default: false,
+        description: "If set, ignores any configured proxy. This will override any proxy environment variables or parameters. This will be set by default if ignore_proxy is set to a value other than 'false' or '0'."
+
+      property :proxy_url, String,
+        description: "Specifies the proxy URL to use during the download."
+
+      property :proxy_user, String,
+        description: "The username to use to build a proxy credential with. Will be consumed by the proxy_credential property if both this property and proxy_password are set"
+
+      property :proxy_password, String,
+        description: "The password to use to build a proxy credential with. Will be consumed by the proxy_credential property if both this property and proxy_user are set"
+
+      load_current_value do
+        current_state = is_choco_installed?
+        current_value_does_not_exist! if current_state == false
+        current_state
+      end
+
+      def is_choco_installed?
+        ::File.exist?("#{ENV["ALLUSERSPROFILE"]}\\chocolatey\\bin\\choco.exe")
+      end
+
+      def define_resource_requirements
+        [ new_resource.proxy_user, new_resource.proxy_password ].each do
+          requirements.assert(:install) do |a|
+            a.assertion do
+              (!new_resource.proxy_user.nil? && new_resource.proxy_password.nil?) || (new_resource.proxy_user.nil? && !new_resource.proxy_password.nil?)
+            end
+            a.failure_message(Chef::Exceptions::ValidationFailed, "You must specify both a proxy_user and a proxy_password")
+            a.whyrun("Assuming that if you have configured a 'proxy_user' you must also supply a 'proxy_password'")
+          end
+        end
+      end
+
+      action :install, description: "Installs Chocolatey package manager" do
+        unless new_resource.download_url.nil?
+          powershell_exec("Set-Item -path env:chocolateyDownloadUrl -Value #{new_resource.download_url}")
+        end
+
+        unless new_resource.chocolatey_version.nil?
+          powershell_exec("Set-Item -path env:chocolateyVersion -Value #{new_resource.chocolatey_version}")
+        end
+
+        if new_resource.use_native_unzip == true
+          powershell_exec("Set-Item -path env:chocolateyUseWindowsCompression -Value true")
+        end
+
+        if new_resource.ignore_proxy == true
+          powershell_exec("Set-Item -path env:chocolateyIgnoreProxy -Value true")
+        end
+
+        unless new_resource.proxy_url.nil?
+          powershell_exec("Set-Item -path env:chocolateyProxyLocation -Value #{new_resource.proxy_url}")
+        end
+
+        if !new_resource.proxy_user.nil? && new_resource.proxy_password.nil? || new_resource.proxy_user.nil? && !new_resource.proxy_password.nil? || !new_resource.proxy_user.nil? && !new_resource.proxy_password.nil?
```

Comment:
Need Karnaugh map for this

Reply:
Running this through, it appears to be a NAND operation?
`if !(new_resource.proxy_user.nil? && new_resource.proxy_password.nil?)`

Reply:
Which...
`unless new_resource.proxy_user.nil? && new_resource.proxy_password.nil?`